File: backend/src/main/java/org/jdk/project/dto/ApiResponse.java
package org.jdk.project.dto;

import lombok.Getter;

@Getter
public class ApiResponse<T> {

  private final int code;
  private final String msg;
  private final T data;

  private ApiResponse(int code, String msg, T data) {
    this.code = code;
    this.msg = msg;
    this.data = data;
  }

  public static <T> ApiResponse<T> ok(T data, String message) {
    return new ApiResponse<>(200, message, data);
  }
}

File: backend/src/main/kotlin/org/jdk/project/dto/ApiResponse.kt
package org.jdk.project.dto

class ApiResponse<T> private constructor(
  val code: Int,
  val msg: String,
  val data: T?,
) {
  companion object {
    @JvmStatic
    fun <T> ok(data: T?, message: String): ApiResponse<T> {
      return ApiResponse(200, message, data)
    }
  }
}
